Sources/SwiftParser/Diagnostics/ParserDiagnosticMessages.swift

```diff
@@ -153,9 +153,9 @@ public struct UnexpectedNodesError: ParserError {
       message += " '\(shortContent)'"
     }
     if let parent = unexpectedNodes.parent {
-      if let parentTypeName = parent.nodeTypeNameForDiagnostics, parent.children(viewMode: .sourceAccurate).first?.id == unexpectedNodes.id {
+      if let parentTypeName = parent.nodeTypeNameForDiagnostics(allowBlockNames: false), parent.children(viewMode: .sourceAccurate).first?.id == unexpectedNodes.id {
         message += " before \(parentTypeName)"
-      } else if let parentTypeName = parent.ancestorOrSelf(where: { $0.nodeTypeNameForDiagnostics != nil })?.nodeTypeNameForDiagnostics {
+      } else if let parentTypeName = parent.ancestorOrSelf(where: { $0.nodeTypeNameForDiagnostics(allowBlockNames: false) != nil })?.nodeTypeNameForDiagnostics(allowBlockNames: false) {
```

Comment:
I wouldn't be against pulling out this into a `firstNodeTypeNameForDiagnostics`, we do this in a bunch of places.

Reply:
I just improved this to

```swift  
let parentTypeName = parent.ancestorOrSelf(mapping: { $0.nodeTypeNameForDiagnostics(allowBlockNames: false) })
```

in one of the commits I created locally today (and which I’ll create a PR for in the next couple of days). I think the new call is sufficiently simple that we don’t need to create a new method for it. WDYT?

Reply:
Sure, makes sense.